matchbox_server/src/signaling.rs

```diff
@@ -453,8 +463,159 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn match_different_id_same_next() {
+        let _ = pretty_env_logger::try_init();
+        let api = api();
+
+        let mut client_a = warp::test::ws()
+            .path("/scope_1?next=2")
+            .handshake(api.clone())
+            // .handshake(ws_echo())
+            .await
+            .expect("handshake");
+
+        let mut client_b = warp::test::ws()
+            .path("/scope_2?next=2")
+            .handshake(api.clone())
+            // .handshake(ws_echo())
+            .await
+            .expect("handshake");
+
+        let mut client_c = warp::test::ws()
+            .path("/scope_1?next=2")
+            .handshake(api.clone())
+            // .handshake(ws_echo())
+            .await
+            .expect("handshake");
+
+        let mut client_d = warp::test::ws()
+            .path("/scope_2?next=2")
+            .handshake(api.clone())
+            // .handshake(ws_echo())
+            .await
+            .expect("handshake");
+
+        client_a
+            .send(Message::text(r#"{"Uuid": "uuid-a"}"#.to_string()))
+            .await;
+        client_c
+            .send(Message::text(r#"{"Uuid": "uuid-c"}"#.to_string()))
+            .await;
+        client_b
+            .send(Message::text(r#"{"Uuid": "uuid-b"}"#.to_string()))
+            .await;
+
+        client_d
+            .send(Message::text(r#"{"Uuid": "uuid-d"}"#.to_string()))
+            .await;
+
+        // Clients should be matched in pairs as they arrive, i.e. a + c and b + d
+        let new_peer_c = recv_peer_event(&mut client_a).await;
+        let new_peer_d = recv_peer_event(&mut client_b).await;
+
+        assert_eq!(new_peer_c, PeerEvent::NewPeer("uuid-c".to_string()));
+        assert_eq!(new_peer_d, PeerEvent::NewPeer("uuid-d".to_string()));
+
+        let timeout = time::sleep(Duration::from_millis(100));
+        pin_mut!(timeout);
+        select! {
+            _ = client_a.recv() => panic!("unexpected message"),
+            _ = client_b.recv() => panic!("unexpected message"),
+            _ = client_c.recv() => panic!("unexpected message"),
+            _ = client_d.recv() => panic!("unexpected message"),
+            _ = &mut timeout => {}
+        }
+    }
+    #[tokio::test]
+    async fn match_same_id_different_next() {
```

Comment:
here we check
- player a -> scope_1?next=2
- player b -> scope_1?next=3
- player c -> scope_1?next=2
- player d -> scope_1?next=3
- player e -> scope_1?next=3